core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -911,17 +919,16 @@ private[spark] class TaskSetManager(
   override def checkSpeculatableTasks(minTimeToSpeculation: Int): Boolean = {
     // Can't speculate if we only have one task, and no need to speculate if the task set is a
     // zombie.
-    if (isZombie || numTasks == 1) {
+    if (isZombie || numTasks == 1 || !speculationEnabled) {
```

Comment:
I don't think you need this change -- whether speculation is enabled is checked before calling this function (https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala#L177)